function/colcon_cd.sh

```diff
@@ -0,0 +1,79 @@
+# copied from colcon_cd/function/colcon_cd.sh
+
+colcon_cd() {
+  if [ $# = 0 ]; then
+    # change the working directory to the previously saved path
+    if [ "$_colcon_cd_root" = "" ]; then
+      echo "No previous path saved. Either call 'colcon_cd <pkgname>' from a" \
+        "directory where <pkgname> can be found or 'colcon_cd --set' to" \
+        "directly save the current working directory." 1>&2
+      return 1
+    fi
+    if [ "$_colcon_cd_root" != "$(pwd)" ]; then
+      cd "$_colcon_cd_root"
+    fi
+
+  elif [ $# = 1 ]; then
+    if [ "$1" = "--set" ]; then
+      # store the current working directory for future invocations
+      _colcon_cd_root="$(pwd)"
+      echo "Saved the current working directory for future invocations of" \
+        "'colcon_cd <pkgname>' as well as to return to using 'colcon_cd'." \
+        "Call 'colcon_cd --reset' to unset the saved path."
+      return 0
+    elif [ "$1" = "--reset" ]; then
+      # unset the save path
+      unset _colcon_cd_root
+      return 0
+    fi
+
+    _colcon_cd_cmd="colcon list --packages-select $1 --paths-only"
```

Comment:
Maybe it's more fragile (I'm not sure how `colcon list` works under-the-hood), but it seems faster to use the following `find` command on my machine, especially when there's multiple locations found:

```
find ${_colcon_cd_root} -name package.xml -execdir grep -q "<name>${1}</name>" {} \; -print | sed s/package.xml//g
```

Though if we can't assume that `_colcon_cd_root` is the root of a colcon workspace, then we might have to adapt the command to ignore the install space:

```
# making a strong assumption about the install location of package.xml
find ${_colcon_cd_root} -name package.xml \! -path ".*/install/${1}/share/${1}/package.xml" -execdir grep -q "<name>${1}</name>" {} \; -print | sed s/package.xml//g
```

Reply:
That approach would only work if your package has a `package.xml` file which isn't the case for all packages supported by `colcon`. Therefore this needs to use `colcon list` to utilize the package discovery and identification logic already present.